tests/end_to_end/capture_helper.hpp

```diff
@@ -28,4 +39,100 @@ class capture_helper_t final
     get_stdout_contents(void);
     std::string
     get_stderr_contents(void);
-};
\ No newline at end of file
+};
+
+// This class can be used by tests to catch stdout and stderr output.
+
+#define TEMPORARY_STDOUT_FILENAME ".\\stdout.txt"
+#define TEMPORARY_STDERR_FILENAME ".\\stderr.txt"
+
+std::string
+capture_helper_t::get_stdout_contents(void)
+{
+    end_capture();
+    return _stdout_contents;
+}
```

Comment:
Putting function definitions in a .hpp causes linker problems when the header file is included by multiple c/cpp files.  These should be left in capture_helper.cpp